src/components/ProteinsForDiseaseCardContainer.tsx

```diff
@@ -1,32 +1,46 @@
-import React, { Fragment, FunctionComponent } from "react";
-import axios from "axios";
-import { withRouter, RouteComponentProps } from "react-router";
-import useApi from "./UseApi";
+import React, { useState } from "react";
+import { v1 } from "uuid";
+import { DropdownButton } from "franklin-sites";
 import { ProteinData } from "./cards/ProteinCard";
-import PageTemplate from "../PageTemplate";
-import { Context } from "../types/context";
-import DiseaseContainer from "./DiseaseContainer";
-import PageContainer from "../PageContainer";
+import ProteinCardCompact from "./cards/ProteinCardCompact";
+import Axios from "axios";
 import { baseUrl } from "../config";
-import ProteinForDiseaseFilterContainer from "./ProteinForDiseaseFilterContainer";
+import { useParams } from "react-router";
+import UseApi from "./hooks/UseApi";
 
-const ProteinsForDiseaseCardContainer: FunctionComponent<
-  RouteComponentProps<any> & { identifier?: string }
-> = ({ match, identifier }) => {
-  let id: string;
-  if (identifier) {
-    id = identifier;
-  } else {
-    id = match.params.id;
+enum Filters {
+  INT = "interactions",
+  PAT = "pathways",
+  DRU = "drugs",
+  SEQ = "variants",
+}
+
+const ProteinForDiseaseCardContainer = () => {
+  const { diseaseid, proteinid } = useParams();
+  const [selectedFilters, setSelectedFilters] = useState({
+    [Filters.INT]: false,
+    [Filters.PAT]: false,
+    [Filters.DRU]: false,
+    [Filters.SEQ]: false,
+  });
+
+  const { data } = UseApi<{ results: ProteinData[] }>(
+    `${baseUrl}/disease/${diseaseid}/proteins`
+  );
+
+  if (!data) {
+    return null;
   }
 
-  const { data, isLoading } = useApi<{ results: ProteinData[] }>(
-    `${baseUrl}/disease/${id}/proteins`
+  const { results } = data;
+
+  const sortedData = results.sort(
```

Comment:
```suggestion
  const sortedData = Array.from(results).sort(
```
To avoid mutating the result in the state of `UseApi`